src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -576,6 +583,9 @@ protected function getKernelParameters()
             'kernel.project_dir' => realpath($this->getProjectDir()) ?: $this->getProjectDir(),
             'kernel.environment' => $this->environment,
             'kernel.debug' => $this->debug,
+            /**
+             * @deprecated since Symfony 4.2
```

Comment:
we would need a way to deprecate DIC parameters, otherwise we would not receive any warning when using it

Reply:
But then again, I don't think `kernel.name` is that useful and I don't think people are using it anyway.